update_data.py

```diff
@@ -380,7 +392,9 @@ def process_request(request, campaign, cursor, table):
         if int(req['total_events']) != int(sample[2]):
             # print('Total events is wrong %s != %s' % (req['total_events'], sample[2]))
             continue
-
+        
+        target_num_events = sample[-1]
```

Comment:
Avoid using global variables in cases like this. It's very hard to track where they come from and how they change. I almost wrote a comment for line `sql_args.append(target_num_events)` that target_num_events comes from nowhere. Maybe it's possible to add this number to `req` dictionary, so it would "travel" together with the request rather than float somewhere around in the global scope.